### PR TITLE
Update vendor of containers/buildah

### PR DESCRIPTION
**Pull request by rhatdan**
Changes since 2022-05-31:
  - add --omit-history option (buildah PR 4028)
  - add --logsplit (buildah PR 4034)

Signed-off-by: Ed Santiago <santiago@redhat.com>
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### Does this PR introduce a user-facing change?
yes

```release-note
podman build now supports the --omit-history option which eliminates the history field from images. 
podman build now supports --logsplit which allows you to specify a directory for build logs when using multiple platforms.  One log for each platform will be generated in the specified directory.

```

### CODE REVIEW

**Comment by vrothberg** [comment]
Please merge https://github.com/containers/podman/pull/14526 before.

**Comment by vrothberg** [comment]
One more rebase.

**Comment by rhatdan** [comment]
Yup I dropped back.  Sadly we don't have this issue on Buildah.

**Comment by edsantiago** [comment]
You can test locally with:
```console
$ env PODBIN_NAME=remote ./test/buildah-bud/run-buildah-bud-tests --filter="omit-history"
```
To test again, either add `--no-checkout` to the above or run `rm -rf test-buildah*`. (The latter is necessary if you've changed the `apply-patches` script. If you've only made changes in the podman tree, just run `make` and use the former).

**Comment by rhatdan** [comment]
@vrothberg @edsantiago I believe this now works except for a fundamental difference between buildah inspect and podman image inspect.

buildah inspect returns data about both Docker{} and OCIv1{} structure, including a History field ({{.Docker.History}} and .OCIv1.History}. Plus the inspect returns a {{.History}} field.  Podman image inspect only returns the .History field.  The buildah check is testing all 3 fields so I can not just turn the test on.  I could go back into Buildah and turn off the check for the other two fields. Or should we enhance Podman image inspect to be closer to buildah inspect?

**Comment by rhatdan** [comment]
```
$ podman inspect 2bda5a2e1724db2b53fc3965085513a3897c9a010f77e94e12148d1e3d2ff991 
[
     {
          "Id": "2bda5a2e1724db2b53fc3965085513a3897c9a010f77e94e12148d1e3d2ff991",
          "Digest": "sha256:18ef078a86ce165621c2f2ec8ab2d9b3a175700c2a5b7a63d9cba583761fd29a",
          "RepoTags": [],
          "RepoDigests": [],
          "Parent": "",
          "Comment": "",
          "Created": "2022-06-13T21:41:17.853958658Z",
          "Config": {
               "Env": [
                    "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
               ],
               "Cmd": [
                    "/bin/sh"
               ],
               "Labels": {
                    "io.buildah.version": "1.27.0-dev"
               }
          },
          "Version": "",
          "Author": "",
          "Architecture": "amd64",
          "Os": "linux",
          "Size": 5817726,
          "VirtualSize": 5817726,
          "GraphDriver": {
               "Name": "overlay",
               "Data": {
                    "LowerDir": "/home/dwalsh/.local/share/containers/storage/overlay/24302eb7d9085da80f016e7e4ae55417e412fb7e0a8021e95e3b60c67cde557d/diff",
                    "UpperDir": "/home/dwalsh/.local/share/containers/storage/overlay/b9e343cf90efb38e4c636260c8175ad0512809f46f2c30d92f2ca625af14c12f/diff",
                    "WorkDir": "/home/dwalsh/.local/share/containers/storage/overlay/b9e343cf90efb38e4c636260c8175ad0512809f46f2c30d92f2ca625af14c12f/work"
               }
          },
          "RootFS": {
               "Type": "layers",
               "Layers": [
                    "sha256:24302eb7d9085da80f016e7e4ae55417e412fb7e0a8021e95e3b60c67cde557d",
                    "sha256:8b964ff2f9efa9791728a25a3e039bda535def729d7e85db9e565ad0186519e7"
               ]
          },
          "Labels": {
               "io.buildah.version": "1.27.0-dev"
          },
          "Annotations": {
               "org.opencontainers.image.base.digest": "sha256:4ff3ca91275773af45cb4b0834e12b7eb47d1c18f770a0b151381cd227f4c253",
               "org.opencontainers.image.base.name": "docker.io/library/alpine:latest"
          },
          "ManifestType": "application/vnd.oci.image.manifest.v1+json",
          "User": "",
          "History": null,
          "NamesHistory": [
               "docker.io/library/61813059c2f9454b9b5e332373c63e88514294d5bd8e1008221bdde28c7a0fe2-tmp:latest"
          ]
     }
]
```
Versus
```
 $ buildah inspect 2bda5a2e1724db2b53fc3965085513a3897c9a010f77e94e12148d1e3d2ff991 
{
    "Type": "buildah 0.0.1",
    "FromImage": "",
    "FromImageID": "2bda5a2e1724db2b53fc3965085513a3897c9a010f77e94e12148d1e3d2ff991",
    "FromImageDigest": "sha256:18ef078a86ce165621c2f2ec8ab2d9b3a175700c2a5b7a63d9cba583761fd29a",
    "Config": "{\"created\":\"2022-06-13T21:41:17.853958658Z\",\"architecture\":\"amd64\",\"os\":\"linux\",\"config\":{\"Env\":[\"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\"],\"Cmd\":[\"/bin/sh\"],\"Labels\":{\"io.buildah.version\":\"1.27.0-dev\"}},\"rootfs\":{\"type\":\"layers\",\"diff_ids\":[\"sha256:24302eb7d9085da80f016e7e4ae55417e412fb7e0a8021e95e3b60c67cde557d\",\"sha256:8b964ff2f9efa9791728a25a3e039bda535def729d7e85db9e565ad0186519e7\"]}}",
    "Manifest": "{\"schemaVersion\":2,\"mediaType\":\"application/vnd.oci.image.manifest.v1+json\",\"config\":{\"mediaType\":\"application/vnd.oci.image.config.v1+json\",\"digest\":\"sha256:2bda5a2e1724db2b53fc3965085513a3897c9a010f77e94e12148d1e3d2ff991\",\"size\":418},\"layers\":[{\"mediaType\":\"application/vnd.oci.image.layer.v1.tar\",\"digest\":\"sha256:24302eb7d9085da80f016e7e4ae55417e412fb7e0a8021e95e3b60c67cde557d\",\"size\":5811200},{\"mediaType\":\"application/vnd.oci.image.layer.v1.tar\",\"digest\":\"sha256:8b964ff2f9efa9791728a25a3e039bda535def729d7e85db9e565ad0186519e7\",\"size\":4608}],\"annotations\":{\"org.opencontainers.image.base.digest\":\"sha256:4ff3ca91275773af45cb4b0834e12b7eb47d1c18f770a0b151381cd227f4c253\",\"org.opencontainers.image.base.name\":\"docker.io/library/alpine:latest\"}}",
    "Container": "",
    "ContainerID": "",
    "MountPoint": "",
    "ProcessLabel": "",
    "MountLabel": "",
    "ImageAnnotations": {
        "org.opencontainers.image.base.digest": "sha256:4ff3ca91275773af45cb4b0834e12b7eb47d1c18f770a0b151381cd227f4c253",
        "org.opencontainers.image.base.name": "docker.io/library/alpine:latest"
    },
    "ImageCreatedBy": "",
    "OCIv1": {
        "created": "2022-06-13T21:41:17.853958658Z",
        "architecture": "amd64",
        "os": "linux",
        "config": {
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/bin/sh"
            ],
            "Labels": {
                "io.buildah.version": "1.27.0-dev"
            }
        },
        "rootfs": {
            "type": "layers",
            "diff_ids": [
                "sha256:24302eb7d9085da80f016e7e4ae55417e412fb7e0a8021e95e3b60c67cde557d",
                "sha256:8b964ff2f9efa9791728a25a3e039bda535def729d7e85db9e565ad0186519e7"
            ]
        }
    },
    "Docker": {
        "created": "2022-06-13T21:41:17.853958658Z",
        "container_config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/bin/sh"
            ],
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "io.buildah.version": "1.27.0-dev"
            }
        },
        "config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "/bin/sh"
            ],
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "io.buildah.version": "1.27.0-dev"
            }
        },
        "architecture": "amd64",
        "os": "linux",
        "rootfs": {
            "type": "layers",
            "diff_ids": [
                "sha256:24302eb7d9085da80f016e7e4ae55417e412fb7e0a8021e95e3b60c67cde557d",
                "sha256:8b964ff2f9efa9791728a25a3e039bda535def729d7e85db9e565ad0186519e7"
            ]
        }
    },
    "DefaultMountsFilePath": "",
    "Isolation": "oci",
    "NamespaceOptions": [
        {
            "Name": "cgroup",
            "Host": false,
            "Path": ""
        },
        {
            "Name": "ipc",
            "Host": false,
            "Path": ""
        },
        {
            "Name": "mount",
            "Host": false,
            "Path": ""
        },
        {
            "Name": "network",
            "Host": false,
            "Path": ""
        },
        {
            "Name": "pid",
            "Host": false,
            "Path": ""
        },
        {
            "Name": "user",
            "Host": false,
            "Path": ""
        },
        {
            "Name": "uts",
            "Host": false,
            "Path": ""
        }
    ],
    "Capabilities": null,
    "ConfigureNetwork": "NetworkDefault",
    "CNIPluginPath": "",
    "CNIConfigDir": "",
    "IDMappingOptions": {
        "HostUIDMapping": true,
        "HostGIDMapping": true,
        "UIDMap": [],
        "GIDMap": []
    },
    "History": null,
    "Devices": null
}
```

**Comment by edsantiago** [comment]
EINSUFFICIENTBRAINPOWER (due to late hour of day). Nice catch. I will think about it in the morning.

**Comment by vrothberg** [comment]
> I could go back into Buildah and turn off the check for the other two fields. Or should we enhance Podman image inspect to be closer to buildah inspect?

I vote for changing the tests.  Changing `podman inspect` could break Docker compat.

**Comment by edsantiago** [comment]
LGTM but one more set of eyeballs please

**Comment by rhatdan** [comment]
@edsantiago @vrothberg I was mistaken, and it looks like omit-history tests are firing and passing against podman -remote, so this PR is ready to merge.

**Comment by edsantiago** [comment]
@mheon could you check the `release-note` block? On reflection it looks to me like this adds a new option (`omit-history`) and hence should include that in release notes?

File: cmd/podman/images/build.go [hunk 1]
```diff
@@ -191,6 +191,7 @@ func buildFlags(cmd *cobra.Command) {
 		_ = flags.MarkHidden("compress")
 		_ = flags.MarkHidden("volume")
 		_ = flags.MarkHidden("output")
+		_ = flags.MarkHidden("logsplit")
```
Comment (vrothberg) [inline cmd/podman/images/build.go:194]: I think we should not hide since it's mentioned already in the Podman man pages.

**Comment by mheon** [comment]
@edsantiago As long as it's a Buildah change, we generally don't include it... My preference instead is to link the Buildah release notes, so that folks can look at those.

That said, `podman build` is sufficiently distinct from Buildah that I would not object if someone did include `podman build` changes in the release notes.

File: cmd/podman/images/build.go (hunk 1 above)
Comment (rhatdan) [reply cmd/podman/images/build.go:194]: This is for podman -remote.

Comment (vrothberg) [reply cmd/podman/images/build.go:194]: Ah, thanks!

**Pull request merged**